File: dotnet/Mcma.Worker/WorkerRequest.cs
using System;
using System.Collections.Generic;
using Mcma.Core;
using Mcma.Core.ContextVariables;
using Mcma.Core.Serialization;
using Newtonsoft.Json.Linq;

namespace Mcma.Worker
{
    public class WorkerRequest : ContextVariableProvider
    {
        public WorkerRequest(string operationName, IDictionary<string, string> contextVariables)
            : base(contextVariables)
        {
            OperationName = operationName;
        }

        public string OperationName { get; }

        public JObject Input { get; set; }

        public T GetInput<T>()
        {
            try
            {
                 return Input != null ? Input.ToMcmaObject<T>() : default(T);
            }
            catch (Exception ex)
            {
                throw new Exception(
                    $"Worker request input could not be deserialized to type {typeof(T).Name}. See inner exception for details.", ex);
            }
        }

        public bool TryGetInput<T>(out T dataObject)
        {
            dataObject = default(T);
            try
            {
                 dataObject = GetInput<T>();

                 return true;
            }
            catch
            {
                return false;
            }
        }
    }
}

File: nodejs/mcma-worker/lib/worker-request.js
const { ContextVariableProvider } = require("mcma-core");

class WorkerRequest extends ContextVariableProvider {
    constructor(request) {
        const operationName = request && request.operationName;
        const contextVariables = request && request.contextVariables;
        const input = request && request.input;

        if (!operationName || typeof operationName !== "string" || operationName.length === 0) {
            throw new Error("operationName must be a non-empty string.");
        }
        
        super(contextVariables);

        this.operationName = operationName;
        this.input = input;
    }
}

module.exports = {
    WorkerRequest
};